fuzz/fuzz_targets/refcounted_model.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2021-2022 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or MIT.
+
+//! Model check with reference counting:
+//! Checks that a sequence of operations and restarts behaves the same as an in-memory collection.
+
+#![no_main]
+use arbitrary::Arbitrary;
+use libfuzzer_sys::fuzz_target;
+use std::collections::btree_map::{BTreeMap, Entry};
+use tempfile::tempdir;
+
+#[derive(Arbitrary, Debug, Clone, Copy)]
+enum Operation {
+	Set(u8, u8),
+	Dereference(u8),
+	Reference(u8),
+}
+
+#[derive(Arbitrary, Debug)]
+enum Action {
+	Transaction(Vec<Operation>),
+	Restart,
+}
+
+#[derive(Arbitrary, Debug, Clone, Copy)]
+enum CompressionType {
+	NoCompression,
+	Snappy,
```

Comment:
We could also have only one compressed variant (Lz4).

Reply:
Indeed. But I'm not sure it harms to test both variants.